### ISSUE TITLE
关于如何使用matlab进行测试

### ISSUE THREAD

**Issue by xiongxiong1996** [WangXueyang-uestc/ARConv#7]
您好，感谢您分享这份优秀的代码，我已经成功按照您的代码，使用WV3数据集完成了模型训练，并得到了 .pth 格式的权重文件。

我注意到测试阶段使用了 MATLAB 脚本，需要依赖 output_mulExm_0.mat 到 output_mulExm_19.mat 这些文件。我在项目中看到您已经将这些 .mat 文
件上传到了百度网盘，但我更希望自己生成这些文件，以便测试自训练的模型效果。

请问我应该如何生成这些 .mat 文件？是通过对训练完成的模型进行推理，并保存推理结果为 .mat 格式吗？我在您提供的代码中暂时没有找到用于生成这些 .mat 文件的部分，不知道是否可以提供相关的生成脚本或说明？这将对我非常有帮助，非常感谢！

期待您的回复！

**Comment by WangXueyang-uestc** [comment]
[MetricCode使用教程.pdf](https://github.com/user-attachments/files/19906958/MetricCode.pdf)

**Comment by xiongxiong1996** [comment]
我想我可能需要参考
DLPan-Toolbox-main\01-DL-toolbox(Pytorch)\UDL\pansharpening\run_test_pansharpening.py
这里给出了得到.mat文件后如何进行测试，我希望可以通过加载模型参数，自己生成这些.mat文件。

**Comment by WangXueyang-uestc** [comment]
对的，你原来说的就是对的，自己将结果保存成mat文件，我给出的代码里没有这一部分

**Comment by xiongxiong1996** [comment]
后面会给公布这部分代码吗？或者有什么相关的代码我可以参考吗？我希望可以复现这部分实验。

**Comment by WangXueyang-uestc** [comment]
model = ARNet().cuda()
model = nn.DataParallel(model)
model.load_state_dict(torch.load(r'G:\PythonProjectGpu\ARConv\modelpara\checkpoint_600_2024-11-05-11-47-04.pth')['model'])
model.eval()
file_path = r"G:\PythonProjectGpu\ARConv\test_data\test_wv3_multiExm1.h5"
test_gt, test_lms, test_ms, test_pan = load_set(file_path)
result_list = []
for i in range(0, 20):
    with torch.no_grad():
        print('------------------------------------------------------')
        print(f'Processing image {i + 1} of 20')
        print('------------------------------------------------------')
        output3 = model(test_pan[i], test_lms[i], 1000, [1, 18])
        # save_img(output3[0], 'output', i)
        ERAS = get_ergas(test_gt[i][0][0:3][...][...], output3[0][0:3][...][...])
        SAM = get_sam(test_gt[i][0][0:3][...][...], output3[0][0:3][...][...])
        eras_list.append(ERAS)
        sam_list.append(SAM)
        output3 = rearrange(output3, 'bs c h w -> bs h w c') * 2047
        result_list.append(output3)
test_gt = rearrange(test_gt.squeeze(1), 'bs c h w -> bs h w c') * 2047
result_list = torch.concatenate(result_list, dim=0)
print(result_list.shape)
print(test_gt.shape)
save_name = os.path.join(
    "D:\\Desktop\\DLPan-Toolbox-main\\02-Test-toolbox-for-traditional-and-DL(Matlab)\\2_DL_Result\\WV3\\ARNet\\results",
    'output_mulExm'+'.mat')
sio.savemat(save_name, {'sr': output3.cpu().numpy(), 'gt': test_gt.cpu().numpy()})

你可以参考这一部分代码，里面有*2047的操作，这是因为我加载数据的时候进行了缩放(如果你用的是我写的data.py,来加载数据
)

**Comment by xiongxiong1996** [comment]
谢谢您！我去尝试一下

**Comment by WangXueyang-uestc** [comment]
不客气，欢迎有问题随时来问
